recipes/nasm/all/conanfile.py

```diff
@@ -11,85 +18,100 @@ class NASMConan(ConanFile):
-                self.run("nmake /f {} {}".format(os.path.join("Mkfiles", "msvc.mak"), " ".join("{}=\"{}\"".format(k, v) for k, v in autotools.vars.items())))
-                shutil.copy("nasm.exe", "nasmw.exe")
-                shutil.copy("ndisasm.exe", "ndisasmw.exe")
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=(self._settings_build.os == "Windows"))
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        if is_msvc(self):
+            VCVars(self).generate()
+            tc.configure_args.append("-nologo")
         if self.settings.arch == "x86":
-            self._autotools.flags.append("-m32")
+            tc.extra_cflags.append("-m32")
         elif self.settings.arch == "x86_64":
-            self._autotools.flags.append("-m64")
-        self._autotools.configure(configure_dir=self._source_subfolder)
+            tc.extra_cflags.append("-m64")
+        tc.generate()
 
-        # GCC9 - "pure" attribute on function returning "void"
-        tools.replace_in_file("Makefile", "-Werror=attributes", "")
+        env = VirtualBuildEnv(self)
+        env.generate()
 
-        # Need "-arch" flag for the linker when cross-compiling.
-        # FIXME: Revisit after https://github.com/conan-io/conan/issues/9069, using new Autotools integration
-        if str(self.version).startswith("2.13"):
-            tools.replace_in_file("Makefile", "$(CC) $(LDFLAGS) -o", "$(CC) $(ALL_CFLAGS) $(LDFLAGS) -o")
+        # TODO I assume this is required
+        # inject requires env vars in build scope
+        # it's required in case of native build when there is AutotoolsDeps & at least one dependency which might be shared, because configure tries to run a test executable
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
 
-        return self._autotools
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        if self.settings.compiler == "Visual Studio":
-            self._build_vs()
+        apply_conandata_patches(self)
+
+        if is_msvc(self):
+            with chdir(self, self.source_folder):
+                self.run("nmake /f {}".format(os.path.join("Mkfiles", "msvc.mak")))
+                shutil.copy("nasm.exe", "nasmw.exe")
+                shutil.copy("ndisasm.exe", "ndisasmw.exe")
         else:
-            autotools = self._configure_autotools()
+            autotools = Autotools(self)
+            autotools.configure()
+
+            # GCC9 - "pure" attribute on function returning "void"
+            replace_in_file(self, "Makefile", "-Werror=attributes", "")
+
+            # Need "-arch" flag for the linker when cross-compiling.
+            # FIXME: Revisit after https://github.com/conan-io/conan/issues/9069, using new Autotools integration
```

Comment:
this issue is closed, is this code still required?

Reply:
I'm not sure, it is hard for me to comment on or test as I don't cross-compile locally.
Also, I see there are a bunch of PRs open on AutoTools so I didn't want to test anything related to AutoTools specifically as whatever problems I hit may already be in the pipeline...